code/modules/security levels/keycard authentication.dm

```diff
@@ -45,9 +45,22 @@ var/global/list/obj/machinery/keycard_auth/authenticators = list()
 			if(active == 1)
 				//This is not the device that made the initial request. It is the device confirming the request.
 				if(event_source)
-					event_source.confirmed = 1
-					event_source.event_confirmed_by = usr
+					confirmed = 1
+
+					// All the useful information is in the source device, so copy it over. There's probably a better way to do this.
+					event_triggered_by = event_source.event_triggered_by
+					event_confirmed_by = usr
+					event = event_source.event
+					ert_reason = event_source.ert_reason
+
+					playsound(src, get_sfx("card_swipe"), 60, 1, -5)
+					to_chat(user, "You swipe your ID card to confirm the [event].")
+
+					trigger_event(event)
 			else if(screen == 2)
+				playsound(src, get_sfx("card_swipe"), 60, 1, -5)
+				to_chat(user, "You swipe your ID card to request the [event].")
```

Comment:
Same as above.

Reply:
Should the existing to_chat stuff be notices as well? Telling you if it isn't powered n stuff

Reply:
Probably better.